# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFT2FontList.h
# Commit: a345e05df151
# Full Hash: a345e05df151563625fe05ec63094b3dae2c9379
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-07 21:31:11
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r 288a67aed661 -r a345e05df151 gfx/thebes/gfxFT2FontList.h
--- a/gfx/thebes/gfxFT2FontList.h	Tue Jul 07 04:35:08 2020 +0000
+++ b/gfx/thebes/gfxFT2FontList.h	Tue Jul 07 04:36:27 2020 +0000
@@ -11,6 +11,9 @@
 #include "gfxPlatformFontList.h"
 
 namespace mozilla {
+
+class CacheAwareZipReader;
+
 namespace dom {
 class SystemFontListEntry;
 };
@@ -174,7 +177,7 @@
   void AppendFacesFromFontFile(const nsCString& aFileName,
                                FontNameCache* aCache, StandardFile aStdFile);
 
-  void AppendFacesFromOmnijarEntry(nsZipArchive* aReader,
+  void AppendFacesFromOmnijarEntry(mozilla::CacheAwareZipReader* aArchive,
                                    const nsCString& aEntryName,
                                    FontNameCache* aCache, bool aJarChanged);
 